### ISSUE TITLE
Typescript Support?

### ISSUE THREAD

**Issue by bayssmekanique** [GoogleChromeLabs/comlink-loader#1]
I see that Clooney has TS support, but this appears to not play well the TypeScript Linter.  Is it even possible to add support for TypeScript given the mutation that occurs?

**Comment by koalalorenzo** [comment]
Any update? That would be really nice!

**Comment by koriwi** [comment]
Dito!

**Comment by developit** [comment]
This is fixed in 2.0.0! There's an example in the README.

**Comment by ahuglajbclajep** [comment]
The README defines `greet` [as follows](https://github.com/GoogleChromeLabs/comlink-loader/blob/2.0.0/README.md#L112-L116), but this code is a [type error](https://www.typescriptlang.org/play/#code/KYDwDg9gTgLgBAQwM4E8B2BjOAzArpmASwjTgHMphgYAKJXAIwCtgMYAuOJGKQtMgJSduvfnADeAWABQcOJRi4opAAYAJYABtNEADRwAJOPrNWMAL4BCFQG4Z5mUA).

```ts
export async function greet(subject: string): string {
  return `Hello, ${subject}!`;
}
```

I think it should be defined as follows.

```ts
export async function greet(subject: string): Promise<string> {
  return `Hello, ${subject}!`;
}
```
